apps/web/src/app/[locale]/challenge/_components/challenge-track-navigation.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { useQuery } from '@tanstack/react-query';
+import { useRouter } from 'next/navigation';
+import { useMemo } from 'react';
+
+import { type ChallengeRouteData } from '~/app/[locale]/challenge/[id]/getChallengeRouteData';
+import { getTrackDetails } from '~/app/[locale]/tracks/_components/track.action';
+
+import { Button } from '@repo/ui/components/button';
+import { Tooltip, TooltipContent, TooltipTrigger } from '@repo/ui/components/tooltip';
+import { cn } from '@repo/ui/cn';
+import { Swords, ChevronLeft, ChevronRight } from '@repo/ui/icons';
+
+import { ChallengeTrackOutline } from './challenge-track-outline';
+
+interface Props {
+  challenge: ChallengeRouteData['challenge'];
+  track: ChallengeRouteData['track'];
+  isCollapsed: boolean;
+  className?: string;
+}
+
+export function ChallengeTrackNavigation({ challenge, track, isCollapsed, className }: Props) {
+  const router = useRouter();
+
+  const { data: trackDetails, isLoading } = useQuery({
+    queryKey: ['track-details', track?.id],
+    queryFn: () => {
+      console.log('fetching track');
+      return getTrackDetails(track!.id);
+    },
```

Comment:
We should remove the `console.log`:
```suggestion
    queryFn: () => getTrackDetails(track!.id),
```